test/reporters/tracer.go

```diff
@@ -0,0 +1,134 @@
+package reporters
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/Microsoft/ApplicationInsights-Go/appinsights"
+	"github.com/Microsoft/ApplicationInsights-Go/appinsights/contracts"
+	"github.com/sirupsen/logrus"
+)
+
+// skip log lines which are just artifacts of ginkgo
+var exclude = []string{"", "S\n", "SS\n", "========================\n", "------------------------------\n", "•\n"}
+
+type TraceWriter struct {
+	c   appinsights.TelemetryClient
+	log *logrus.Entry
+
+	w *io.PipeWriter
+	r *io.PipeReader
+
+	br *bufio.Reader
+}
+
+func (tw *TraceWriter) reader() {
+	for {
+		s, err := tw.br.ReadString('\n')
+		if !stringInSlice(s, exclude) {
+			tw.c.TrackTrace(s, contracts.Information)
+		}
+		if err != nil {
+			break
+		}
+	}
+}
+
+func (tw *TraceWriter) WaitForEmtpyBuffer() error {
```

Comment:
I didn't do a good job of writing this first time around, please re-pull from https://gist.github.com/jim-minter/7464be56dd1bfbc0be21ab5821ac68d5
- removes the waiting, implements cleanup properly
- removes the unnecessary pipe in TraceWriter